mibidata/color.py

```diff
@@ -252,28 +245,33 @@ def compose_overlay(image, overlay_settings):
         if len(overlay_settings) == 1 and 'channels' in v:
             overlay_settings = v['channels']
             break
-        raise ValueError("Unexpected format of overlay_settings dictionary.")
+        raise ValueError('Unexpected format of overlay_settings dictionary.')
 
     overlay = None
     for channel in overlay_settings:
-        item = overlay_settings[channel]
+        setting = overlay_settings[channel]
+        array = image[channel]
+        # If set to min brightess, skip this channel:
+        if setting['brightness'] == constants.OVERLAY_MIN_BRIGHTNESS:
+            continue
         array = image[channel]
         # Because we treat the min differently, don't use np.clip
-        range_min = item.get('intensity_lower', 0)
-        range_max = item.get('intensity_higher', array.max())
+        range_min = setting.get('intensity_lower', 0)
+        range_max = setting.get('intensity_higher', array.max())
         array[array > range_max] = range_max
         array[array < range_min] = 0
         array = array / float(range_max)
-        array[array > 0] += item.get('brightness', 0)
-        # Apply default filter even if no blurring
-        array = ndimage.filters.convolve(array, kernel)
-        if item.get('blur', 0) > 0:
-            ndimage.filters.gaussian_filter(
-                array, item['blur'] * 100, output=array)
+        array = ndimage.filters.convolve(
+            array, constants.OVERLAY_SMOOTHING_KERNELS[setting['blur']])
         np.clip(array, 0, 1, out=array)
+        if setting['brightness'] > 0:
+            array /= (1 - setting['brightness'])
+            np.clip(array, 0, 1, out=array)
+        elif setting['brightness'] < 0:
```

Comment:
how come brightness has different effects for positive (linear) and negative (non-linear) values?

Reply:
also this math would seem to support the fact that a max_range=10 and brightness 0 should be equivalent to max_range=1 and brightness 0.9, correct? I would not expect similar equivalency if brightness<0.